### PR TITLE
Adds construction bags for engineers

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Adds construction bags for engineers (starting in the engineering personal closets).
They can hold stock parts, other common machine components, circuits, misc electronics and cable coils.

## Why It's Good For The Game

Lets engineers do construction without having to dedicate half their inventory on components for a single machine. Made half-obsolete later by RPEDs, but being able to store multiple circuitboards is still useful.
I really want to encourage machine construction from engineering, since they even have access to advanced machinery like quantum pads but i never see them make use of it.

## Changelog
:cl:
add: Engineering closets now start with a construction bag, which can hold machine components, circuits, electronics, and cable coils.
/:cl:

### CODE REVIEW

**Comment by partykp** [comment]
Maybe add them to the vending machine instead, just like chembags

**Comment by XDTM** [comment]
Oh, good idea

**Comment by Detective-Google** [comment]
Looks really good, maybe a quick image on the PR body would help (so I don't have to look at the diff bot for the sprite :p)

**Comment by XDTM** [comment]
It's not great tbh, but here it is
immagine

**Comment by nemvar** [comment]
Very cool. I will be perfectly honest, I hate this. I hate this mentality from the buttom of my being. It's easy, it's braindead, it's overpowered, but we are willing to except it just because it's convinient right now. Everybody loves a good little powercreep here and there. But this isn't good long term. Why do we even have inventory limits if we are just going to add a storage bag that can keep infinite amounts whenever we need it? I will ask you this, do you know the weight class of a melon? A huge fuck off melon? It's WEIGHT_CLASS_TINY / 100. At least botany, chemistry and xenobio have an excuse. They have to carry huge amounts of stuff regularly, but as an engineer, just use a fucking box. Why can't you people not get that in your head? Maybe there's a reason why weight classes even exist? Maybe you are supposed to work around inventory constraints? This is just LOOOOL I want 450 cloning circuit boards in my box, because why the fuck not? Yeah why the fuck not? I will fucking tell you why, it's because if you want a rewards, you have to god damn work for it.

This is a huge slipperly slope, maybe not tomorrow, but soon we will have people that will say: "Why can't we put metal and glass and plasteel and wood and surgery tools and whatever bullshit we also want to carry infinite amounts of into the engineering bag. Why don't we have roundstart security bags that carry 50 guns and have weight class tiny?". I'm calling this, it will happen. That's the nature of powercreep.

**Comment by Dorsisdwarf** [comment]
I hate it when people are blatantly trying to become a copypasta

**Comment by XDTM** [comment]
I'm thinking of the long term here, though. The point isn't to be a mindless buff to engineers, so they can do what they already do more easily. It's to make a certain playstyle, namely machine construction, simpler and thus more appealing, since i rarely see it done when spectating. By fostering this playstyle indirectly i hope to see more machine construction, in particular experimental machines who aren't found roundstart (like quantum pads), leading to more frequent interaction between those and the crew, thus leading to more varied/interesting rounds depending on how they are used.

**Comment by nemvar** [comment]
If you really want to foster machine construction, add more machines worth building, make the underlying system less cluncky, add spaces that encourage people to build machines. For a somewhat freeform game, there barely is any place anywhere to actually allow the players to express themselves.

The thought process behind this isn't braindead and I don't recall saying that. But this does make the actual act of playing require less thought. This is misguided.

**Comment by XDTM** [comment]
The main reason for the system's clunkyness is having to carry a lot of stock parts, hence my solution being a bag to carry them. It's pretty much the same as light replacers (which didn't ruin the game nor slippery sloped).

I don't think that there is a loss in the thought involved; it changes machine construction from "too annoying to even bother" to "actually doable, let's think of what we can build and where"

**Comment by nemvar** [comment]
If you stated goal is to get people to build quantum pads, why PR a feature that literally powercreeps them at the same time?

What you added isn't a fix, it's a bandaid. You added a feature that invalides a working system (inventory management) to fix one that is cluncky and broken (construction).

The big limiter with construction never was carrying space, you already got tools like the RPED and boards really don't take that much space if you are smart about it. It was that you only had two sources of boards on the entire station and have to run across the whole thing if you miss just one thing. If you really want to fix the part problem, remove the concept of individiual parts entirely. Add an item that directly fabricates parts from materials and can be upgraded with design disks, or something like that. Just like the light replacer did it.

In short, powercreeping bandaids bad. We need modern solutions to our problems.

**Comment by SpaceManiac** [comment]
There's already RPEDs for stock parts, and circuit boards can be put in a standard box to store more of them. I'm not against encouraging construction but I'm not sure this really does that.

**Comment by MCterra10** [comment]
Can we please give these to engiborgs so they can finally build machines on their own? I hate begging humans to insert circuit boards into machine frames

**Comment by Cyprex** [comment]
What nemvar said. 30-50 metal, some glass, a box with circuit boards already fit easily in your backpack, an RPED can be carried in your backpack. However, if I dart off to botany to construct a chem dispenser and forget one of the many components a chem dispenser needs I have to go all the way back to engineering. The current system just makes me fulfill a checklist of items, where it's easy to miss some items. That said, should construction be easy? Why even bother with other departments if you can just make your own chemistry/robotics/medbay... setup in engineering in a reasonable amount of time. Point being, some machines shouldn't be easily available while others should. Making construction easier with our current systems doesn't seem like a good idea to me.

**Comment by XDTM** [comment]
I agree that a construction rework might be better in the long run, but in the meantime i'm using this as a "bandaid" until there's a definite idea on how construction should be done.

Also, as far as i know, RPEDs do not fit into a backpack until they're bluespace, which makes them really annoying to constantly carry around.

**Comment by FloranOtten** [comment]
As an engineering main, I dislike this. Inventory management is a skill which takes a bit of fumbling to learn but is certainly not impossible. Circuit boards go in the box (You can wear the breath mask and replace the TRay scanner in your pocket with the oxygen tank). I rarely use more than 6 boards at once. For parts; Maybe make the basic rped fit in the backpack. It is severely annoying to carry it around, which coupled with the constraints of only 50 parts makes it bad.

**Comment by FantasticFwoosh** [comment]
Well this solves the problem of me carrying around a duffelbag everywhere as engineer.
 
> As an engineering main, I dislike this. Inventory management is a skill which takes a bit of fumbling to learn but is certainly not impossible. 

I disagree, its a pain to be so attention intensive to sort your inventory out for doing your job and repeat return visits to pick up usually single bits you need at a time, the only complaint i may have is individuals taking entire stocks of stuff and walking off, or standing over the autolathe for long periods of time.

**Comment by XDTM** [comment]
@FloranOtten Making RPEDs fit in backpacks would be kinda the same as this PR. The point of the construction bags is to carry stock parts, them carrying circuits doesn't affect inventory management since you could already fit them in boxes.

Also, remember the point of the PR is to make construction less annoying. It _doesn't matter_ if you can already inventory manage your stuff into being able to carry construction materials; apparently it's enough of a hassle that right now most engineers never build any new machinery.

**Comment by Mickyan** [comment]
Re: storage powercreep

The backpack full of boxes combo is what's the problem with storage powercreep, specialized bags should be the way forward

Though I think this bag should be WEIGHT_CLASS_BULKY so you can only put it on your belt, engineers should decide if they want faster access to their tools or their materials

**Comment by XDTM** [comment]
It's still a slot in your backpack that is only going to be useful if you plan on building machines with it. If people use this every round it means people are actually building stuff, which is what i want in the first place.

**Comment by Mickyan** [comment]
nevermind me I thought this also allowed steel/glass stacks

**Review by 81Denton — APPROVED** [review]
